refactor(volunteer): add new create volunteer form

BaxLi opened the pull request:
<details>
<summary>Screenshots</summary>

![localhost_4200_admin_volunteers_list](https://user-images.githubusercontent.com/1881266/104103397-8e490480-52aa-11eb-8866-9a19ba4fbd05.png)
</details>

Closes #146

BaxLi committed:
go for PR

BaxLi committed:
checked - push to PR

BaxLi committed:
New Request form & selection address & icons

BaxLi commented:
Added new request form and selection address component ...

unrevised6419 commented on src/app/admin/volunteers/volunteers-list/volunteers-list.component.html:
```diff
@@ -47,6 +73,34 @@ <h2>Voluntari</h2>
   </form>
 </mat-card>
 
+<mat-card
+  class="mb-4 d-flex flex-column flex-sm-row flex-wrap"
+  style="
+    justify-content: space-between;
+    height: 40px;
+    font-size: 14px !important;
+    align-content: center;
+  "
+>
+  <div>
+    <span class="ml-1">Calendar</span>
+    <mat-icon style="transform: scale(0.4)">fiber_manual_record</mat-icon>
+    <span class="ml-1">Disponibilitate Voluntari</span>
+  </div>
+  <mat-icon
+    mat-button
+    style="
+      font-size: 16px !important;
+      height: 16px !important;
+      width: 16px;
+      font-weight: bold;
+      text-align: center;
+    "
+  >
+    add
+  </mat-icon>
+</mat-card>
+
```
This block is not needed at the moment.

Commit:
refactor(dialog): extract no padding dialog class

Commit:
refactor(volunteers): remove not needed calendar preview placeholder

iamandrewluca commented on src/app/admin/requests/request-details/request-details.component.html:
```diff
@@ -1,22 +1,295 @@
-<div class="mt-4">
-  <a href="#" appGoBack>&lt;- Înapoi</a>
+<div class="modal_header_line" fxLayout="row" fxLayoutAlign="center center">
+  <a>Crează Cerere Nouă</a>
```
Why page for showing request details, was refactored into page to create request?

unrevised6419 commented on src/app/admin/requests/request-details/request-details.component.html:
```diff
@@ -1,22 +1,295 @@
-<div class="mt-4">
-  <a href="#" appGoBack>&lt;- Înapoi</a>
+<div class="modal_header_line" fxLayout="row" fxLayoutAlign="center center">
+  <a>Crează Cerere Nouă</a>
```
Why page for showing request details, was refactored into page to create request?

BaxLi replied:
The component was totally rewrote to cover issue #145. According to the Figma,  New Component and Modify existing one functionality have the same view component. So it take sense to have one component for both cases (mean request-details component), and, inside *.ts cover difference between logic of create new request (issue145) and modify existent request (issue not-exist).

BaxLi replied:
Start to modify according phone discussion

unrevised6419 commented on src/app/admin/requests/request-details/request-details.component.html:
```diff
@@ -1,22 +1,295 @@
+    </div>
+  </div>
 
-<app-request-form
```
`app-request-form` usage was removed, but the component was not removed, remained a lot of dead code

BaxLi replied:
Suppose this component will be be used later for viewing  existing requests.

BaxLi replied:
now we start re-engineer the code where <app-request-form> will be used for both new record and existing one - so it will remain ...

unrevised6419 commented on src/app/admin/volunteers/shared/form-hours-selector/form-hours-selector.component.ts:
```diff
@@ -0,0 +1,98 @@
+import { MatFormFieldControl } from '@angular/material/form-field';
+
+@Component({
+  selector: 'app-form-hours-selector',
```
> custom component for hours-selection as in figma

Sorry but this component does not look at all how is in figma :|

![image](https://user-images.githubusercontent.com/1881266/103589831-01422c00-4ef5-11eb-95a4-9fdc7fd4e7d0.png)

BaxLi replied:
Sorry - according to discussion with Andrey Valcov it was accepted to change such view instead of this one:
![image](https://user-images.githubusercontent.com/4039092/103620527-505f7f80-4f3c-11eb-9025-97eaec6a7c95.png)
to this one. I will ask for Andrei confirmation and attach it here.

BaxLi replied:
so the (figma owner) answer 
Andrei Valcov : @Salut, Este OK asa cum ai facut.@

unrevised6419 commented on src/app/admin/volunteers/shared/newvolunteer-register-form/newvolunteer-register-form.component.scss:
```diff
@@ -0,0 +1,160 @@
+$gapBetweenFieldAndHint: "25px";
+$fontFamily: Roboto, sans-serif;
+
+$modalWindowWidth: 555px;
+$modalWindowHeight: 881px;
```
No hardcoded widths and heights, UI should be responsive

BaxLi replied:
sorry - but it is properties from Figma 
![image](https://user-images.githubusercontent.com/4039092/103621633-2dce6600-4f3e-11eb-9b30-f79397867b30.png)

BaxLi replied:
changed to
$modalWindow_maxWidth: 555px;
$modalWindow_maxHeight: 881px;

![image](https://user-images.githubusercontent.com/4039092/103669683-562b8400-4f81-11eb-84b2-3913108a7b92.png)

unrevised6419 commented on src/app/admin/volunteers/shared/newvolunteer-register-form/newvolunteer-register-form.component.scss:
```diff
@@ -0,0 +1,160 @@
+$gapBetweenFieldAndHint: "25px";
+$fontFamily: Roboto, sans-serif;
```
font-family is set globaly

BaxLi replied:
$fontFamily: Roboto, sans-serif; - removed as requested.

unrevised6419 commented on src/app/admin/volunteers/shared/newvolunteer-register-form/newvolunteer-register-form.component.scss:
```diff
@@ -0,0 +1,160 @@
+  background-color: rgb(255, 255, 255);
+}
+
+mat-icon {
```
this affects all other icons in dialog, for example if you want to add a dialog close icon

BaxLi replied:
Yes - its according our vision. So, since <mat-icon> style is not defined globally according to the Figma properties,  we define such one for an exact component item (newvolunteer-register-form) inside scss, to avoid usage of inline styles

unrevised6419 commented on src/app/shared/constants.ts:
```diff
@@ -26,14 +26,44 @@ export const KIV_ZONES_MAP = KIV_ZONES.reduce((acc, zone) => {
   operator = 'operator',
 }
 
+export enum ZONES {
```
Is this a duplicate of `KIV_ZONES` above?

BaxLi replied:
Yes, back-end uses constants in enum format. So the global approach looks like Front-End and Back-End should use the same defined constants. Theoretically frontEnd may import such constants via a API call. That's why we tried to align constants usage to the Back-End approach.

BaxLi replied:
as discussed and accepted by phone - this enum comply with back-end.

unrevised6419 commented on src/app/admin/volunteers/volunteers-list/volunteers-list.component.ts:
```diff
@@ -163,10 +169,10 @@ export class VolunteersListComponent implements OnInit {
-      maxWidth: '100%',
-      maxHeight: '90vh',
+    const dialogRef = this.matDialog.open(NewVolunteerRegisterFormComponent, {
+      width: '555px',
+      height: '881px',
```
Please no hardcoded dialog sizes, it should be responsive based on content

BaxLi replied:
I will discuss with Andrei Valcov to comply with Figma

BaxLi replied:
As requested changed to: 
maxWidth: '50%',
height: '90vh',

unrevised6419 commented on src/app/admin/volunteers/shared/newvolunteer-register-form/newvolunteer-register-form.component.html:
```diff
@@ -0,0 +1,168 @@
+          <img
+            src="/assets/icons/volunteers/{{
+              volunteerRolesIncons[role.key]
+            }}.svg"
```
Svg icons should be registered in `app.component.ts` and used properly using `<mat-icon svgIcon="icon-name"></mat-icon>`

BaxLi replied:
agree - will change as wrote

BaxLi replied:
Done

unrevised6419 commented on src/app/admin/volunteers/shared/newvolunteer-register-form/newvolunteer-register-form.component.scss:
```diff
@@ -0,0 +1,160 @@
+
+.footer {
+  border-top: 1px solid #ccc;
+  padding-top: 23px;
```
Please no magic numbers.

BaxLi replied:
changed - switch to a variable ... 
$footerPaddingTopBottom: 23px;

.footer {
  border-top: 1px solid #ccc;
  padding-top: $footerPaddingTopBottom;

BaxLi committed:
Merge branch 'dev' of https://github.com/code4moldova/voluntar-web into For_PR_Branch

BaxLi committed:
PR-correction 1(non final)

BaxLi committed:
SVG Icons registred in App level

BaxLi committed:
small errors by eslynth adjusted

Commit:
Merge branch 'dev' into For_PR_Branch

Commit:
refactor(volunteers): remake create volunteer dialog

- Removed hours dialog, used 2 selects instead
- Added string localizations
- Renamed dialog component
- Removed all CSS
- Fixed some constants

Commit:
refactor(requests): restore requests to dev version, split in another PR
